packages/fast-breakpoint-tracker-react/src/breakpoint-tracker.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from "react";
+import { canUseDOM } from "exenv-es6";
+import { throttle } from "lodash-es";
+import { Breakpoints, defaultBreakpoints, identifyBreakpoint } from "./breakpoints";
+
+export interface BreakpointTrackerProps {
+    /**
+     * The render method
+     */
+    render: (activeBreakpoint: keyof Breakpoints | void) => React.ReactNode;
+}
+
+export interface BreakpointTrackerState {
+    activeBreakpoint: keyof Breakpoints | void;
+}
+
+export class BreakpointTracker extends React.Component<BreakpointTrackerProps, BreakpointTrackerState> {
+    /**
+     * The current breakpoint.
+     */
+    public static breakpoints: Breakpoints = defaultBreakpoints;
+
+    /**
+     * Track if we have an open animation frame request
+     */
+    private openRequestAnimationFrame: boolean;
+
+    /**
+     * Constructor for the BreakpointTracker component.
+     */
+    constructor(props: BreakpointTrackerProps) {
+        super(props);
+
+        this.state = {
+            activeBreakpoint: null
+        };
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidMount(): void {
+        if (canUseDOM()) {
```

Comment:
Is there a reason you can't do this work in the constructor? It seems like you can prevent an initial re-render if you move this work there.

Reply:
Added a comment, to ensure we address this. Added below as well:

We are doing this work in a lifecycle method instead of the constructor to ensure that server rendered instances with conditional DOM rendering by breakpoint will be re-rendered. ReactDOM.hydrate() will call the constructor again, but it does not trigger a re-render. It will only bind event handlers. The only way to ensure the correct DOM is consistently rendered on the client is to perform this work here.

So, if I have a view that conditionally renders DOM based on breakpoint, and I run that on the server, the HTML that is output may be different than what is actually required. When ReactDOM.hydrate() is called, the event listeners will be bound, but it will not re-render, meaning we could have an inaccurate DOM.

Reply:
Makes sense, thanks for adding the comment